datafusion/core/src/avro_to_arrow/arrow_array_reader.rs

```diff
@@ -861,6 +861,7 @@ fn flatten_string_values(values: &[&Value]) -> Vec<Option<String>> {
     values
         .iter()
         .flat_map(|row| {
+            let row = maybe_resolve_union(row);
```

Comment:
We must do this here, to match the logic used to deduce the offsets. Without this the offsets would not match the flattened values.